### PR TITLE
feat: s3 replication config

### PR DESCRIPTION
**Pull request by patheard**
# Summary
Add the ability to create s3 bucket replication rules.

Also removes the hardcoded bucket names from the s3 tests to prevent failures related to bucket names already being taken.

### CODE REVIEW

**Review by dinophile — APPROVED** [review]
🍅

File: S3/input.tf [hunk 1]
```diff
@@ -83,3 +83,9 @@ variable "object_lock_configuration" {
   type        = any
   default     = {}
 }
+
+variable "replication_configuration" {
+  description = "(Optional) Map containing cross-region replication configuration."
+  type        = any
+  default     = {}
+}
```
Comment (CalvinRodo) [inline S3/input.tf:91]: Do we need the level of configuration this gives us? 

Do we have a use case for this other than replicate everything in a bucket into another bucket in the same region?

Comment (CalvinRodo) [reply S3/input.tf:91]: Should this just be the ARN of the destination bucket for simplicity's sake?

**Review by CalvinRodo — CHANGES_REQUESTED** [review]
Please see my comments, if the requirement is for more complicated replication scenarios i'll happily approve. But ideally this would be as simple as possible.

File: S3/input.tf (hunk 1 above)
Comment (patheard) [reply S3/input.tf:91]: For cloud based sensor, I'll need to control encryption with a KMS key and changing object ownership.

Things like `metrics` and `replication_time` could be dropped with no problem.  I could also wipe out the `filters` and just use full object replication (default to a blank filter).

Comment (CalvinRodo) [reply S3/input.tf:91]: Yes I'd like to keep it as simple as possible, we can always add in the complexity if needed later.

**Review by CalvinRodo — COMMENTED** [review]
:+1:

**Pull request merged**